tests/types/configs.test-d.cts

```diff
@@ -0,0 +1,18 @@
+import configs = require("@eslint-community/eslint-plugin-eslint-comments/configs")
+import expectTypeModule = require("expect-type")
+
+import type { Linter } from "eslint"
+
+import expectTypeOf = expectTypeModule.expectTypeOf
```

Comment:
Should most of this not be `const` instead of `import`? 
```suggestion
const configs = require("@eslint-community/eslint-plugin-eslint-comments/configs")
const { expectTypeOf } = require("expect-type")

import type { Linter } from "eslint"
```

---

I have never seen this syntax before, so do correct me if I am wrong :sweat_smile:

Reply:
In this file I'm testing the types to make sure they work with TypeScript's CJS syntax. So something like **`const configs = require("@eslint-community/eslint-plugin-eslint-comments/configs")`** becomes **`import configs = require("@eslint-community/eslint-plugin-eslint-comments/configs")`**. It's to test ESM/CJS Interoperability on a type level.

[Reference](https://www.typescriptlang.org/docs/handbook/modules/reference.html#export--and-import--require)

Reply:
Oh, neat! Thanks for the explanation : )

Reply:
@scagood It might be good to fully migrate to TypeScript so the types and runtime code don't get out of sync. Thoughts?

Reply:
Here is the previous discussion in the `eslint-plugin-n` repo, I am okay with either tbh
https://github.com/eslint-community/eslint-plugin-n/issues/371

---


That being said, that is more a question for @eslint-community/eslint-plugin-eslint-comments.